doc/release/1.14.0-notes.rst

```diff
@@ -200,15 +208,12 @@ is eventually written back to the original array. For ``UPDATEIFCOPY``, the
 writeback to the original array occurs at ndarray deallocation. For
 ``WRITEBACKIFCOPY``, a new ``PyArray_ResolveWritebackIfCopy`` function must be
 explicitly called before deallocation (or, if an error occurred,
-``PyArray_DiscardWritebackIfCopy``). Numpy now mostly uses the
-``WRITEBACKIFCOPY`` mechanism internally, but still uses ``UPDATEIFCOPY`` in
-nditer use for backward-compatibility with the python nditer interface.
-
-In python code, if numpy is compiled with ``-DDEPRECATE_UPDATEIFCOPY`` or if
-run on PyPy, ``DeprecationWarning`` warnings will be issued on use of
-``UPDATEIFCOPY`` (eg, when nditer is used) if writeback resolution has not
-occurred before calling ``array_dealloc``. In the future this warning will
-always be issued, once nditer stops using ``UPDATEIFCOPY``.
+``PyArray_DiscardWritebackIfCopy``). Numpy now uses the
+``WRITEBACKIFCOPY`` mechanism internally.
+
+``DeprecationWarning`` warnings will be issued on use of
+``UPDATEIFCOPY`` or if writeback resolution has not
+occurred before calling ``array_dealloc``.
```

Comment:
If `array_dealloc` sees `WRITEBACKIFCOPY` then that's just a bug, right? Shouldn't we issue a `RuntimeWarning` or something? (Really it ought to raise an exception, but since you can't do that from `array_dealloc` a noisy warning is a reasonable alternative.)

Reply:
fixed in 7e5c2a7e6, warning message adjusted to indicate the root cause could be nditer